plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -105,13 +102,16 @@ internal class PubCommand(private val flutterAbsolutePath: File) : CommandLineTo
     override fun transformVersion(output: String) = output.removePrefix("Dart SDK version: ").substringBefore(' ')
 
     override fun command(workingDir: File?): String =
-        if (flutterAbsolutePath.isDirectory) "$flutterAbsolutePath${File.separator}$dartCommand" else dartCommand
+        when {
+            flutterAbsolutePath?.isDirectory == true -> "$flutterAbsolutePath${File.separator}$dartCommand"
+            else -> dartCommand
+        }
 
     private fun commandPub(): String = "${command()} pub"
 
     override fun run(workingDir: File?, vararg args: CharSequence): ProcessCapture {
         var result = ProcessCapture(workingDir, *commandPub().splitOnWhitespace().toTypedArray(), *args)
-        if (result.isError) {
+        if (result.isError && flutterAbsolutePath != null) {
```

Comment:
I'm trying to understand the distinction between the following two code paths:

1. exit from throw in line 123
2. return from line 127 without executing the code block starting in line 114.

In particular, I wonder why the `throw` is needed, and if it is good to enhance the if condition in line 114, or whether it should be enhanced in line 116. What do you think?

If we decide for 114 (as-is), would it then also make sense to drop the `throw`?